linalg/hypre.cpp

```diff
@@ -3539,6 +3539,10 @@ void HypreSmoother::SetOperator(const Operator &op)
       ones = 1.0;
       A->Mult(ones, diag);
    }
+   else if (type == 6)
+   {
+      Z = new HypreParVector(*A);
+   }
```

Comment:
For what version of hypre was this necessary? Looking at v2.29.0, it seems like `Z` is not used for type 6 on CPU, however, it is used on GPU. Was on GPU where you saw segfaults?

Reply:
Yes, they used `Z` on CPU in Hypre 2.20, but anymore not from 2.21. Still, it is used for GPUs.

Reply:
The rabbit hole goes much further, checking the code, `Z` was used at multiple other places in v2.20, which we had not covered. From v2.21, we are missing it for GPUs in many cases, which partially overlap, but not fully to not make things easy 😩 . So should I care about v2.20? And should I fix that GPUs?

Reply:
Ok, how about now? But the list might grow as they implement more cases on GPUs, so it might be better to allocate it always for GPUs for future compatibility 🤔

Reply:
Ok, inverted the logic to allocate Z always on GPUs except the specified cases 😉